microprofile/WFLY-17510-remove-microprofile-opentracing.adoc

```diff
@@ -0,0 +1,100 @@
+
+=== Other Interested Projects
+
+OpenTelemetry/MicroProfile Telemetry are involved but unchanged.
+
+=== Relevant Installation Types
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [ ] Managed domain
+* [x] OpenShift s2i
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+MicroProfile OpenTracing will be migrated to an admin-only module
+
+A `MigrateOperation` will be added to remove the extension from existing server configs upon server startup (in admin-only mode). For systems with MicroProfile OpenTracing configured, OpenTelemetry will be configured using default values.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+Converting MicroProfile OpenTracing config to OpenTelemetry config. The configuration options and default values vary enough that a meaningful conversion is not possible.
+
+== Backwards Compatibility
+
+This represents the removal of a subsystem, so any application the depends on these APIs will fail to deploy. Likewise, any installation that is expecting (Jaeger) tracing data to be exported will see a behavior change that may or may not impact any external monitoring solutions.
+
+All other subsystems should be unaffected by this removal.
+
+=== Default Configuration
+
+The default configuration for MicroProfile configurations will be altered to remove the MicroProfile OpenTracing subsystem. Likewise, the relevant Galleon layers will also have the extension removed from their definitions.
+
+=== Importing Existing Configuration
+
+With the change of the subsystem to admin-only, existing configurations will fail to start in normal standalone modes. Such configurations will need to be started in admin mode to allow for the altering of the configuration, or the extension and subsystem will need to be removed manually.
+
+=== Deployments
+
+Deployments that use MicroProfile OpenTracing APIs directly will fail to deploy. These applications will need to modified to remove this dependency.
+
+Applications that do not use the affected API directly should deploy and run without modification or issue.
+
+Any deployment environments that are expecting implicit tracing information to be exported will be affected. Since OpenTracing and OpenTelemetry differ in the means by which traces are exported, and the trace data itself differs, administrators will need to make the appropriate adjustments to both the configuration of WildFly, as well as any systems are consuming the exported trace data.
+
+=== Interoperability
+
+See the section of above for a discussion of trace data formats and exports.
+
+== Test Plan
+
+Tests will be added to verify that MicroProfile OpenTracing has been removed, and parsing tests will be modified as needed to verify that the system can continue to read/parse the subsystem config during migration.
+
+Tests for MP OpenTracing under `testsuite/integration/microprofile` and `testsuite/integration/microprofile-tck` will be removed.
```

Comment:
And similarly for the EAP MicroProfile TS, see https://github.com/jboss-eap-qe/eap-microprofile-test-suite

Reply:
Added